cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/plugin/PluginInstantiator.java

```diff
@@ -218,14 +232,55 @@ private PluginProperties substituteMacroFields(Plugin plugin, MacroEvaluator mac
     for (Map.Entry<String, PluginPropertyField> pluginEntry : pluginPropertyFieldMap.entrySet()) {
       if (pluginEntry.getValue().isMacroSupported()) {
         String macroValue = plugin.getProperties().getProperties().get(pluginEntry.getKey());
-        properties.put(pluginEntry.getKey(), macroParser.parse(macroValue));
+        if (macroValue != null) {
+          macroValue = macroParser.parse(macroValue);
+        }
+        properties.put(pluginEntry.getKey(), macroValue);
+      } else {
+        properties.put(pluginEntry.getKey(), plugin.getProperties().getProperties().get(pluginEntry.getKey()));
```

Comment:
can move this outside the if block, `plugin.getProperties().getProperties().get(pluginEntry.getKey()` and name it as propertyValue, since its used in both if and else block.